### PR TITLE
SDIT-905 VSIP created rooms have parent and not based on name

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/AgencyInternalLocation.kt [hunk 1]
```diff
@@ -31,6 +31,10 @@ data class AgencyInternalLocation(
   @Convert(converter = YesNoConverter::class)
   val certifiedFlag: Boolean = false,
 
+  @Column(name = "TRACKING_FLAG")
+  @Convert(converter = YesNoConverter::class)
+  val trackingFlag: Boolean = false,
+
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/AgencyInternalLocation.kt:37]: required so room is setup exactly like other Visit rooms

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/AgencyInternalLocation.kt [hunk 2]
```diff
@@ -58,38 +62,10 @@ data class AgencyInternalLocation(
 
   @Column(name = "CAPACITY")
   val capacity: Int? = null,
-) {
-  val isCell: Boolean
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/AgencyInternalLocation.kt:62]: none of this was used, not sure why it was ever added

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/AgencyInternalLocationRepository.kt [hunk 3]
```diff
@@ -7,25 +7,8 @@ import java.util.Optional
...
   fun findByDescriptionAndAgencyId(description: String, agencyId: String): AgencyInternalLocation?
-  fun findOneByLocationId(locationId: Long): Optional<AgencyInternalLocation>
-  fun findByLocationCodeAndAgencyId(locationCode: String, agencyId: String): List<AgencyInternalLocation>
-  fun findByAgencyIdAndLocationTypeAndActiveAndParentLocationIsNull(
-    agencyId: String,
-    locationType: String,
-    active: Boolean,
-  ): List<AgencyInternalLocation>
+  fun findByAgencyIdAndActiveAndLocationCodeInAndParentLocationIsNull(agencyId: String, active: Boolean = true, locationCodes: List<String> = listOf("VISIT", "VISITS")): AgencyInternalLocation?
 }
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/AgencyInternalLocationRepository.kt:14]: The meat - the parent of all visits is defined as:

- A top level location - e.g. no parent
- Have the code either VISIT or VISITS
- Is active

Only 2 prisons do not satisfy this - so will fallback to rooms being created with no parent

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/visits/VisitService.kt [hunk 4]
```diff
@@ -542,11 +546,17 @@ class VisitService(
...
         locationType = "VISIT",
         locationCode = roomCode,
         active = true,
         userDescription = userDescription.uppercase(),
+        capacity = 99,
+        listSequence = 99,
+        trackingFlag = true,
+        currentOccupancy = 0,
+        certifiedFlag = false,
       ),
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/visits/VisitService.kt:560]: setup exactly the same as other visit rooms

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**